### PR TITLE
Docs: Add changelogger info to monorepo.md

### PR DESCRIPTION
**Pull request by samiff**
#### Changes proposed in this Pull Request:

* Add changelogger info to `monorepo.md`
* Update `writing-a-good-changelog-entry.md` with changelogger info; removing info about the previously used `Needs Changelog` labeling/template.

#### Jetpack product discussion

* Internal: p1HpG7-bqa-p2

#### Does this pull request change what data or activity we track or use?

* No.

#### Testing instructions:

* Proofread updated documents.

### CODE REVIEW

**Review by jeherve — APPROVED** [review]

File: docs/monorepo.md [hunk 1]
```diff
@@ -228,3 +228,46 @@ If you have set `.extra.mirror-repo`, `.extra.release-branch-prefix`, and `.extr
...
 * `tools/deploy-to-svn.sh` will prepare a temporary directory with the content of the mirror repo branch that is ready to be pushed to WordPress.org SVN.
 * `tools/revert-release.sh` will prepare a temporary directory that updates the "Stable version" tag in `readme.txt` to the previous version, in case an emergency rollback is required.
+
+## Jetpack Changelogger
+
+The [Jetpack Changelogger](https://packagist.org/packages/automattic/jetpack-changelogger) tool helps in managing a changelog for a project by having each PR drop a specially-formatted "change file" into a changelog directory, which the tool can then process for a release.
+
+As implemented by the Jetpack Monorepo, any PR that touches the Jetpack plugin itself, or anything else in the `/projects` directory will require a specially-formatted file to the project's specified `changelog` directory; there is a [command](#using-the-jetpack-changelogger) mentioned below that can help create the change file.
```
Comment (anomiex) [inline docs/monorepo.md:236]: The grammar is off in this sentence. Maybe (from your edit to the field guide page)
```suggestion
As implemented by the Jetpack Monorepo, any PR that touches the Jetpack plugin itself, or anything else in the `/projects` directory will need to add a specially-formatted file to the project's specified `changelog` directory; there is a [command](#using-the-jetpack-changelogger) mentioned below that can help create the change file.
```

File: docs/monorepo.md [hunk 2]
```diff
@@ -228,3 +228,46 @@ If you have set `.extra.mirror-repo`, `.extra.release-branch-prefix`, and `.extr
...
+
+The “Significance” header specifies the significance of change in the style of [semantic versioning](https://semver.org/): patch, minor, or major.
+
+The “Type” header categorizes the change in the changelog. In Jetpack, for example, our changelog divides changes into “Major Enhancements”, “Enhancements”, “Improved compatibility”, and “Bugfixes”.
+
+The body is separated from the headers by a blank line, and is the text that actually goes into the changelog. This should follow our recommendations for [writing a good changelog entry](./writing-a-good-changelog-entry.md). Feel free to (sparingly) use Markdown in the body text.
+
+### Using the Jetpack Changelogger
+
+The changelogger tool is dev-required by each project via Composer, and after `composer install`, may be run to interactively add a change file using:
```
Comment (anomiex) [inline docs/monorepo.md:255]: Extraneous comma
```suggestion
The changelogger tool is dev-required by each project via Composer, and after `composer install` may be run to interactively add a change file using:
```
Or else a missing comma
```suggestion
The changelogger tool is dev-required by each project via Composer, and, after `composer install`, may be run to interactively add a change file using:
```

File: docs/writing-a-good-changelog-entry.md [hunk 3]
```diff
@@ -1,24 +1,22 @@
...
 
-If you've changed something that users will notice, like improving the display of a block or fixing a bug that prevents a site feature from working, then by all means suggest a changelog entry.
+If you've changed something that users will notice, like improving the display of a block or fixing a bug that prevents a feature from working, then by all means submit a suggested changelog entry.
 
-If you've improved our CI or development environments, or refactored old code without user-visible changes, or something like that, it's probably not something that's worth mentioning in the changelog. But don't worry, we still appreciate it! And remember to say so in the appropriate section of the PR template, don't just leave it blank.
+If you've improved our CI or development environments, or refactored old code without user-visible changes, it's probably not a change that needs a changelog entry intended for end users. However, you would still submit a changelog entry with a ["Comment" header](./monorepo.md#using-the-jetpack-changelogger) which would be omitted from the generated changelog, but still serve as documentation to other developers.
```
Comment (anomiex) [inline docs/writing-a-good-changelog-entry.md:11]:
```suggestion
If you've improved our CI or development environments, or refactored old code without user-visible changes, it's probably not a change that needs a changelog entry intended for end users. However, you would still include a change file with a ["Comment" header](./monorepo.md#using-the-jetpack-changelogger) which would be omitted from the generated changelog, but still serve as documentation to other developers.
```

**Review by anomiex — COMMENTED** [review]
A few minor suggestions

**Review by anomiex — APPROVED** [review]

**Pull request merged**